service/matching/taskListManager.go

```diff
@@ -323,6 +345,10 @@ func (c *taskListManagerImpl) GetTask(
 	ctx context.Context,
 	maxDispatchPerSecond *float64,
 ) (*InternalTask, error) {
+	if c.shouldReload() {
```

Comment:
I think this is a nice compromise with regard to feature enablement, though I would definately add a line in your implementation ERD About this choice. 

If you return with errNoTasks, I imagine the poller behaviour is to immediately re-poll, is that correct?

Reply:
Yes, and it has no effect on our SLA